pkgs/development/ruby-modules/gem-config/default.nix

```diff
@@ -804,6 +804,15 @@ in
       "--with-sqlite3-include=${sqlite.dev}/include"
       "--with-sqlite3-lib=${sqlite.out}/lib"
     ];
+  } // lib.optionalAttrs stdenv.cc.cc.isClang {
+    patches = [
```

Comment:
Patches are preferred to be applied unconditionally so that it is easier to detect when they break. Also `stdenv.cc.isClang` should be fine